yaptide/simulation_runner/shieldhit_runner.py

```diff
@@ -92,26 +93,68 @@ def run_shieldhit(param_dict):
             with open(abs_input_path, 'w') as temp_input_file:
                 temp_input_file.write(input_dict[config_filename])
 
+        print(os.listdir(tmp_output_path))
         settings = SimulationSettings(input_path=tmp_output_path,
                                       simulator_exec_path=None,
                                       cmdline_opts='')
 
-        runner_obj = Runner(jobs=param_dict['jobs'],
-                            keep_workspace_after_run=False,
-                            output_directory=tmp_output_path)
+        runner_obj = SHRunner(jobs=param_dict['jobs'],
+                              keep_workspace_after_run=False,
+                              output_directory=tmp_output_path)
 
         start_time = timeit.default_timer()
-        isRunOk = runner_obj.run(settings=settings)
-        if not isRunOk:
+        if not runner_obj.run(settings=settings):
             return None
 
         elapsed = timeit.default_timer() - start_time
         print("MC simulation took {:.3f} seconds".format(elapsed))
 
-        estimator = runner_obj.get_data()
-
-        return estimator
-
-
-if __name__ == '__main__':
-    sys.exit(run_shieldhit(sys.argv[1:]))
+        estimator_list = runner_obj.get_data()
+
+        return convert_output(estimator_list)
+
+
+def convert_output(estimator_list):
+    """Function for converint simulation output to dictionary"""
+    if not estimator_list:
+        return {"result": "None"}
+
+    result_dict = {"estimators": []}
+    for est_name in estimator_list:
+
+        estimator_dict = {
+            "name" : est_name,
+            "pages": []}
+        for page in estimator_list[est_name].pages:
```

Comment:
What if the data dimension is different than one ?
Code below has several flaws:
 - it violates DRY principle
 - it doesn't work for 0-D data (a single number) - in such case it is adding same number 5 times
 - it doesn't work for 2-D data

If you are planning to add a simple version of the converter which works with 1-D data, then I suggest to:
 - use `page.dimension` property (I assume you know how to access python property) to check if data dimension is 1, see https://github.com/DataMedSci/pymchelper/blob/master/pymchelper/page.py#L48
 - in case dimension is different than 1, print some warning and return nothing
 - in case dimension is 1, the use `page.plot_axis(0)` to get relevant axis. Before doing so, read carefully docstring and code of this method https://github.com/DataMedSci/pymchelper/blob/master/pymchelper/page.py#L123
 - for data extraction you can use `page.data_raw` or `page.data.flatten()` (assuming you know what numpy flatten is doing). For inspiration check how converter to PNG files is working: https://github.com/DataMedSci/pymchelper/blob/master/pymchelper/writers/plots.py#L208